src/mappings/entities/mapPortfolio.ts

```diff
@@ -50,15 +75,26 @@ const handlePortfolioCreated = async (
   const number = getNumberValue(rawPortfolioNumber);
   const name = getTextValue(rawName);
 
-  await createPortfolio(
-    {
-      identityId: ownerId,
-      number,
+  const portfolio = await Portfolio.get(`${ownerId}/${number}`);
```

Comment:
Shoudln't this check go into the `createPortfolioIfNotExists` function, and then we wouldn't need this check here?

If the event is `PortfolioCreated` we shouldn't have it in the DB.

Reply:
People can send to Portofolios that don't exist? Maybe we do need this check after all.

Reply:
> Shoudln't this check go into the createPortfolioIfNotExists function

Yes you are right. That would also prevent the method being misused for cases where Portfolio actually existed

> If the event is PortfolioCreated we shouldn't have it in the DB.

For that I added the check that in case Portfolio got inserted because of Instruction created event before being created, we can then just update the Portfolio name and updatedBlockId

> People can send to Portofolios that don't exist? Maybe we do need this check after all.

This is even more likely to happen instead of DID not being present. Like lets say if you and I had a discussion where I was sending you some ACME tokens and you said that send it to my Portfolio 3 (but that didn't existed). So I created the instruction and then you realised I need to create it to receive this. This would be handled now as well.